CRM/Utils/String.php

```diff
@@ -1034,6 +1038,7 @@ public static function parseOneOffStringThroughSmarty($templateString) {
     }
     $smarty = CRM_Core_Smarty::singleton();
     $cachingValue = $smarty->caching;
+    set_error_handler([$smarty, 'handleSmartyError'], E_USER_ERROR);
```

Comment:
```suggestion
    $originalErrorHandler = set_error_handler([$smarty, 'handleSmartyError'], E_USER_ERROR);
```

Reply:
Then just before the return to set_error_handler($originalErrorHandler, E_USER_ERORR);

Reply:
Or restore_error_handler() I think is better. But I'm not clear on why smarty errors are being limited to this function. Why can't we catch errors in all smarty code?

Reply:
@demeritcowboy this function covers all the smarty errors that might come from user content - anything else would be hit while developing. We could catch those in the form class if we wanted but the only reported time where uncaught smarty errors is an issue is when users report them.

Reply:
@seamuslee001 @demeritcowboy - I've added the `restore_error_handler`